spanner/google/cloud/spanner/streamed.py

```diff
@@ -34,11 +56,18 @@ class StreamedResultSet(object):
         :class:`google.cloud.proto.spanner.v1.result_set_pb2.PartialResultSet`
         instances.
 
+    :type restart: callable
+    :param restart:
+        Function (typically curried via :func:`functools.partial`) used to
+        restart the initial request if a retriable error is raised during
+        streaming.
+
     :type source: :class:`~google.cloud.spanner.snapshot.Snapshot`
     :param source: Snapshot from which the result set was fetched.
     """
-    def __init__(self, response_iterator, source=None):
+    def __init__(self, response_iterator, restart, source=None):
```

Comment:
You don't want a nice default here? Or you don't expect users to interact with this constructor?

Reply:
The latter, really:  instances of this class are returned from `Snapshot.read` and `Snapshot.execute_sql`.